framework/src/modules/chain/index.js

```diff
@@ -85,10 +85,15 @@ module.exports = class ChainModule extends BaseModule {
 
 	async load(channel) {
 		this.chain = new Chain(channel, this.options);
+		const isLiskReady = await channel.invoke('lisk:isLiskReady');
 
-		channel.once('lisk:ready', () => {
+		if (isLiskReady) {
```

Comment:
Not the right way to do it, `lisk:ready` event is triggered after `load` is called. So it must be listened as of previous implementation. I suspect you changed this because of module reload, if yes then as I told earlier, we should develop that strategy as an enhancement later. And leave that implication as it was before.

Reply:
Yes, it was introduced to allow modules to be reloaded in case it crashes and I don't see the reason of reverting this implementation if it works.

Reply:
If child process cause some memory leaks crash, then this infinite loading will exhaust the resources.

Reply:
I agree and we plan to improve this already. This is the first implementation of child process and wont be the final one. Knowing that this implementation can be unstable, I've created a property in the config file to completely disable it if required.

Reply:
I think this change is not related to stability. If there is an issue and any of code crash the whole application exited. That's exactly the default behaviour we have from start and we should keep it consistent in this release. In next release we develop this resilience with a detailed and extensive approach.